### ISSUE TITLE
unparsed connection strings can't handle uri-encoded params like passwords

### ISSUE THREAD

**Issue by dpsutton** [metabase/metabase#14836]
jdbc/query can be given a connection string but it will then parse it. It however, won't correct url-decode the parameters like passwords.

## repro

```emacs-lisp
;; from emacs before startup
(setenv "MB_DB_CONNECTION_URI" "postgresql://localhost:5432/metabase?user=user&password=password")
```

or 
`MB_DB_CONNECTION_URI=postgresql://localhost:5432/metabase?user=user&password=password`

but use a password that requires uri encoding.

### Background
https://clojure.atlassian.net/browse/JDBC-170

had wanted to pass the connection string through untouched by us so we could support more options per each database [issue #14678](https://github.com/metabase/metabase/issues/14678) fixed by [pr #14716](https://github.com/metabase/metabase/pull/14716). It was assumed that passing the connection string through would leave it untouched. However, `clojure.java.jdbc` will actually parse this behind the scenes and doesn't url-decode the parameters. So if your connection string was `"jdbc:postgresql://localhost:5432/database-to-check-env?user=mb_user_test&password=password%26%25%3F"` (url encoded `"password&%?"`) the un-decoded password was used in the db-spec map. A workaround for this is to pass `(jdbc/query {:connection-uri the-url-encoded-connection})`.

Docstring from `jdbc/get-connection`:
```
   Raw:
    :connection-uri (required) a String
                 Passed directly to DriverManager/getConnection
                 (both :user and :password may be specified as well, rather
```

and the code inside that function that we want:

```clojure
     connection-uri
     (-> (if (and user password)
           (DriverManager/getConnection connection-uri user password)
           (DriverManager/getConnection connection-uri))
         (modify-connection opts))
```

**Comment by flamber** [comment]
Should this be kept or closed (and milestone removed)?

**Comment by camsaul** [comment]
This was fixed before shipping 0.38.0
